### DISCUSSION TITLE
Why are the requests automatically redirected 307?

### DISCUSSION THREAD

**Discussion by garbe-github-support** [fastapi/fastapi#9850] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, HTTPException
from ml_base import *

# Pydantic提供了一种简单方便的方法来验证和操作数据
# Web 开发中有句话叫做：永远不要相信用户的输入
# 所以，我们可能需要检查、校验这些数据，将这些数据格式化、标准化，以至于这些数据满足我们真实程序的数据输入，保证项目的正确执行
# FastAPI会自动使用Pydantic模型进行请求体的JSON解析和验证。如果数据无效，FastAPI会自动返回一个有详细错误信息的 HTTP 422 响应
from pydantic import BaseModel
from demo import MakeupTransfer

app = FastAPI()

class Image:
   ....


# POST路由，接收两张图像的url
@app.post("/makeupTransfer/")
async def makeupTransfer(images: Images):
    .....


if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=53445)
```

### Description

Why are the requests sent automatically redirected? I am a novice to FastAPI and simply use FastAPI. The code is as follows, without any other configuration. Why is it automatically redirected, without any log output or other available information.

I try many other port，come to the same thing

**Comment by garbe-github-support** [comment]
My request info
![image](https://github.com/tiangolo/fastapi/assets/131808107/ef6daf6c-4b8b-4845-9c00-60e955f9415c)
![image](https://github.com/tiangolo/fastapi/assets/131808107/f27718cd-efce-4172-97ce-0e69a2ee4c30)
the response 
![image](https://github.com/tiangolo/fastapi/assets/131808107/87e071f2-4cac-4d6e-a41f-4c10b3035384)

**Comment by jgould22** [comment, marked as answer]
By default FastAPI will redirect requests without a `/` at the end to a URL with a slash at the end.

You can see it in the location header of your response there.

If you do no this behaviour fastapi will issue a 404 if you do not have a slash at the end. 

You can disable this behaviour with this setting I believe https://github.com/tiangolo/fastapi/blob/master/fastapi/applications.py#L65C9-L65C25

**Comment by garbe-github-support** [comment]
Okay, thank you very much
